src/messaging/README.md

```diff
@@ -0,0 +1,90 @@
+# Message Layer
+
+## Debug Prints
+
+When progress logging is enabled, the message layer emits a number of prints
+that provide more details on the messages being sent or received. Most of these
+are fairly self explanatory with the exception of the ones listed below.
+
+### Message Transmission
+
+#### Legend
+
+On message transmission, the following print is emitted. The various fields that
+will be expanded are denoted with `$` .
+
+Unless specified, numerical values are represented in decimal notation.
+
+```
+<<< [E:$exchange_id M: $message_id (Ack: $ack_message_id)] ($msg_category) Msg TX to $fabric_index:$destination [$compressed_fabric_id] --- Type $protocol_id:$msg_type ($protocol_name:$message_type_name)
+```
+
+| Field                | Description                                                                                                                            |
+| -------------------- | -------------------------------------------------------------------------------------------------------------------------------------- |
+| exchange_id          | Exchange ID + IsInitiator flag from message header ('i' if initiator, 'r' if responder')                                               |
+| message_id           | Message counter                                                                                                                        |
```

Comment:
Perhaps consistently compress all instances of `message` always as `msg`?

Reply:
+1